### PR TITLE
[quant][pt2e] Add support for channel in DerivedQuantizationSpec

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107833

Summary:
att

Test Plan:
python test/test_quantization.py TestQuantizePT2E.test_derived_qspec_per_channel

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D48630535](https://our.internmc.facebook.com/intern/diff/D48630535)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48630535).

**Review by andrewor14 — APPROVED** [review]

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 1]
```diff
@@ -903,6 +903,124 @@ def validate(self, model: torch.fx.GraphModule) -> None:
...
+                            quant_max=255,
+                            qscheme=torch.per_tensor_affine,
+                            is_dynamic=False,
+                            observer_or_fake_quant_ctr=observer.default_observer,
+                        )
+                        weight_qspec = QuantizationSpec(
+                            dtype=torch.int8,
+                            quant_min=-128,
+                            quant_max=127,
+                            qscheme=torch.per_tensor_affine,
```
Comment (kimishpatel) [inline test/quantization/pt2e/test_quantize_pt2e.py:932]: what you really want to test is have weight qspec with per_channel_affine and then have derived_qparams_fn, return per channel scale tensor. That way the bias zp and scale tensors are of size = output channels

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 2]
```diff
@@ -903,6 +903,124 @@ def validate(self, model: torch.fx.GraphModule) -> None:
...
+                        )
+
+            def validate(self, model: torch.fx.GraphModule) -> None:
+                pass
+
+        m = TestHelperModules.ConvWithBNRelu(relu=False, bn=False).eval()
+        example_inputs = (torch.randn(1, 3, 5, 5),)
+
+        # program capture
+        m, guards = torchdynamo.export(
```
Comment (kimishpatel) [inline test/quantization/pt2e/test_quantize_pt2e.py:980]: not this export, right?

**Review by kimishpatel — COMMENTED** [review]
Sorry got me a bit to review this but I think there are bugs here that we should fix. If it is per channel we should make sure qparams, scale, zp are per channel

File: test/quantization/pt2e/test_quantize_pt2e.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/pt2e/test_quantize_pt2e.py:932]: ah sorry, I'll update it in the capture diff

File: test/quantization/pt2e/test_quantize_pt2e.py (hunk 2 above)
Comment (jerryzh168) [reply test/quantization/pt2e/test_quantize_pt2e.py:980]: capture diff is not landed yet

**Pull request closed without merging**